### PR TITLE
Remove deprecated `Flatten` and `Tile` `Op`s

### LINKED ISSUE #399: Remove deprecated `Flatten` and `Tile` `Op`s
The [`Flatten`](https://github.com/pymc-devs/aesara/blob/f2ecc2e52cb704cacc1b4642fb0bf9c849bd975f/aesara/tensor/basic.py#L2859) and [`Tile`](https://github.com/pymc-devs/aesara/blob/f2ecc2e52cb704cacc1b4642fb0bf9c849bd975f/aesara/tensor/basic.py#L3090) `Op`s are deprecated and need to be removed.

### PR DESCRIPTION
**Pull request by hectormz**
This PR closes #399 and removes long deprecated `Flatten` and `Tile` `Op`s.

I'm not sure if `unittest.rst` is maintained, but I made a change to the code there.

My dev environment is not quite right, so I couldn't confirm if the debug output in `debug_faq.rst` (line 158) needed to be updated.

Here are a few important guidelines and requirements to check before your PR can be merged:
+ [X] There is an informative high-level description of the changes.
+ [X] The description and/or commit message(s) references the relevant GitHub issue(s).
+ [X] [`pre-commit`](https://pre-commit.com/#installation) is installed and [set up](https://pre-commit.com/#3-install-the-git-hook-scripts).
+ [X] The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [X] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes), and there are **no commits that fix changes introduced by other commits in the same branch/BR**.
+ [ ] There are tests covering the changes introduced in the PR.

Don't worry, your PR doesn't need to be in perfect order to submit it.  As development progresses and/or reviewers request changes, you can always [rewrite the history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_rewriting_history) of your feature/PR branches.

If your PR is an ongoing effort and you would like to involve us in the process, simply make it a [draft PR](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests).

### CODE REVIEW

File: aesara/tensor/basic_opt.py [hunk 1]
```diff
@@ -2673,39 +2672,6 @@ def local_useless_split(fgraph, node):
             return [out2]
 
 
-@register_canonicalize
-@register_stabilize
-@local_optimizer([Flatten])
-def local_flatten_lift(fgraph, node):
```
Comment (ricardoV94) [inline aesara/tensor/basic_opt.py:2679]: We should check if we have an equivalent rewrite for reshape

Comment (ricardoV94) [reply aesara/tensor/basic_opt.py:2679]: NVM, I see we already do

**Review by ricardoV94 — APPROVED** [review]
Look good to me. Need to fix a merge conflict

**Comment by hectormz** [comment]
@ricardoV94 Thanks for the review! I merged in changes that @brandonwillard made after my PR

**Comment by ricardoV94** [comment]
I think you will have to rebase from main instead of merging. Can you confirm @brandonwillard?

**Comment by hectormz** [comment]
I can do that as well and re-push

**Comment by hectormz** [comment]
Are you able to restart the checks? Seemed to cancel with force push, but not restart

**Comment by ricardoV94** [comment]
> Are you able to restart the checks? Seemed to cancel with force push, but not restart

I rebased on main again, tests should rerun now

**Review by brandonwillard — APPROVED** [review]
Looks good; much appreciated!

**Pull request merged**